Codescene.VSExtension.VS2022/Codescene.VSExtension.VS2022/Application/ErrorHandling/Logger.cs

```diff
@@ -4,41 +4,59 @@
 using Microsoft.VisualStudio.Shell;
 using System;
 using System.ComponentModel.Composition;
+using System.IO;
 using System.Threading.Tasks;
 using static Codescene.VSExtension.Core.Application.Services.Util.Constants;
 
 namespace Codescene.VSExtension.VS2022.Application.ErrorHandling;
 
+/// <summary>
+/// Contains methods for logging messages in the Visual Studio extension.
+/// Messages are logged to both the output pane and a local log file.
+/// Log rotation is implemented to manage file size and backups.
+/// File names are rotated when they exceed a specified size limit (10 MB).
+/// File logs are stored in the local application data directory under "Codescene".
+/// File name includes the extension version for easy identification (codescene-vs-extension-<version>.log).
+/// </summary>
 [Export(typeof(ILogger))]
 [PartCreationPolicy(CreationPolicy.Shared)]
-internal class Logger : ILogger
+public class Logger : ILogger
 {
     private readonly OutputPaneManager _outputPaneManager;
+    private static readonly string LogFileName = "codescene-vs-extension-" + Vsix.Version + ".log";
+    private static readonly string LogFilePath = Path.Combine(
+        Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+        "Codescene",
+        LogFileName);
+    private static readonly object _fileLock = new object();
+    private const long MaxLogFileSizeBytes = 10 * 1024 * 1024; // 10 MB
+    private const int MaxBackupFiles = 3;
```

Comment:
Could potentially be in the constants file, also naming convention is violated.
```suggestion
    private const long MAX_LOG_FILE_SIZE_BYTES = 10 * 1024 * 1024; // 10 MB
    private const int MAX_BACKUP_FILES = 3;
```